pkg/upstream/http.go

```diff
@@ -31,6 +31,7 @@ var SignatureHeaders = []string{
 	"Date",
 	"Authorization",
 	"X-Forwarded-User",
+	"X-Forwarded-Group",
```

Comment:
I'll need to defer to @JoelSpeed on this one. Technically this is a breaking change which would mean this PR needs to wait until v7.

But I have a feeling no one is using this functionality... https://github.com/oauth2-proxy/oauth2-proxy/issues/648

So do we rip this out and make it non-breaking (but have a very minor security issue where the groups portion won't be a signed header, leaving it open to spoofing).

Or leave it in for utmost security and allow a breaking change cause really, I've seen a few PRs break this functionality and not mention it in Breaking Changes notes.

Reply:
Okies will wait on this one!

Reply:
I think @NickMeves is right, we have never noted this as a breaking change and no one has ever complained about it. It feels bad to me to allow this through as non breaking now that we have had the discussion that it is though 😅 

I'm a bit torn tbh

Reply:
Should we just remove it from the list of headers and make an important note (that way previous functionality will remain working & secure until a user attempts to use signed headers + new group header)?

Then we just overhaul this in V7 or V8? Or at least add this change above in a mini-PR targetted for V7 with the breaking aspect of the change?

Reply:
@stefansedich Lets get the SignatureHeader stuff ripped out and into another PR that we can mark as breaking.

We'll either handle it then or by then I'll rip it all out and refactor it.

Otherwise we'll need to tag this as a `breaking` change and this won't get merged for a couple of months until our next major version update.

Reply:
Gotcha makes sense! have removed that!